Formula/guile.rb

```diff
@@ -35,26 +41,38 @@ class Guile < Formula
   depends_on "pkg-config" # guile-config is a wrapper around pkg-config.
   depends_on "readline"
 
+  ### Remove after configure.ac patch is not necessary anymore.
+  uses_from_macos "flex" => :build
+  ###
+
   uses_from_macos "gperf"
   uses_from_macos "libffi", since: :catalina
   uses_from_macos "libxcrypt"
 
+  on_macos do
+    # A patch to fix JIT on Apple Silicon is embedded below, this fixes:
+    #   https://debbugs.gnu.org/cgi/bugreport.cgi?bug=44505
+    # The patch has been accepted upstream.
+    #   https://git.savannah.gnu.org/cgit/guile.git/commit/?id=3bdcc3668fd8f9a5b6c9a313ff8d70acb32b2a52
+    #  We should remove it from here once Guile 3.0.9 is released.
+    patch :DATA
+  end
+
   def install
     # Avoid superenv shim
     inreplace "meta/guile-config.in", "@PKG_CONFIG@", Formula["pkg-config"].opt_bin/"pkg-config"
 
-    system "./autogen.sh" unless build.stable?
-
-    # Disable JIT on Apple Silicon, as it is not yet supported
-    # https://debbugs.gnu.org/cgi/bugreport.cgi?bug=44505
-    extra_args = []
-    extra_args << "--enable-jit=no" if Hardware::CPU.arm?
+    if OS.mac?
+      # We need to regenerate ./configure because we just patch configure.ac.
+      system "autoreconf", "-vif"
+    else
+      system "./autogen.sh" unless build.stable?
```

Comment:
Why move this?

Reply:
That was because `autogen.sh` is not distributed in the tarball.

Reply:
Sorry, or did you mean why remove `extra_args`? If so, it's because we don't want to disable JIT anymore on arm64 platforms since that's what the patch fixes.